CMakeLists.txt

```diff
@@ -1,94 +1,133 @@
-
-include(CheckIncludeFileCXX)
-
-check_include_file_cxx(any HAS_ANY)
-check_include_file_cxx(string_view HAS_STRING_VIEW)
-check_include_file_cxx(coroutine HAS_COROUTINE)
-if (NOT "${CMAKE_CXX_STANDARD}" STREQUAL "")
-    # Do nothing
-elseif (HAS_ANY AND HAS_STRING_VIEW AND HAS_COROUTINE)
-    set(CMAKE_CXX_STANDARD 20)
-elseif (HAS_ANY AND HAS_STRING_VIEW)
-    set(CMAKE_CXX_STANDARD 17)
-else ()
-    set(CMAKE_CXX_STANDARD 14)
-endif ()
+cmake_minimum_required(VERSION 3.16)
+project(MachiKoroCPP 
+    VERSION 0.0.1
+    LANGUAGES CXX
+)
+
+# ---------------------------------------------------------------------------------------
+# Set project options.
+# ---------------------------------------------------------------------------------------
 
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-add_executable(${PROJECT_NAME} main.cc
-)
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release")
+endif ()
+
+message(STATUS "Started CMake for ${PROJECT_NAME} v${PROJECT_VERSION}...\n")
+
+# ---------------------------------------------------------------------------------------
+# Package dependency.
+# ---------------------------------------------------------------------------------------
 
-# ##############################################################################
-# If you include the drogon source code locally in your project, use this method
-# to add drogon
-# add_subdirectory(drogon)
-# target_link_libraries(${PROJECT_NAME} PRIVATE drogon)
-#
-# and comment out the following lines
 find_package(Drogon CONFIG REQUIRED)
-target_link_libraries(${PROJECT_NAME} PRIVATE Drogon::Drogon)
-
-# ##############################################################################
-
-if (CMAKE_CXX_STANDARD LESS 17)
-    # With C++14, use boost to support any, string_view and filesystem
-    message(STATUS "use c++14")
-    find_package(Boost 1.61.0 REQUIRED)
-    target_link_libraries(${PROJECT_NAME} PUBLIC Boost::boost)
-elseif (CMAKE_CXX_STANDARD LESS 20)
-    message(STATUS "use c++17")
-else ()
-    message(STATUS "use c++20")
-endif ()
+find_package(GTest)
+
+if (NOT GTest_FOUND)
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY  https://github.com/google/googletest.git
+        GIT_TAG         v1.14.0
+    )
+    FetchContent_MakeAvailable(googletest)
+endif (NOT GTest_FOUND)
+
+# ---------------------------------------------------------------------------------------
+# Headers & Sources.
+# ---------------------------------------------------------------------------------------
+
+include(cmake/sources.cmake)
+
+# Model Library
+
+add_library(Model STATIC "")
+target_sources(Model
+    PRIVATE
+    ${MODEL_SOURCES}
+    PUBLIC
+    ${MODEL_HEADERS}
+)
+target_include_directories(Model
+    PUBLIC
+    ${CMAKE_SOURCE_DIR}/models
+    ${CMAKE_SOURCE_DIR}/models/card
```

Comment:
I'm thinking maybe we can just use `target_include_directories(Model PUBLIC ${CMAKE_SOURCE_DIR}/models)`.
In this way, in Model layer, we can use `models` as the root directory for the includes like `#include "card/card.h"` or `#include "hand.h"`.